### PR TITLE
[ALLUXIO-1509] Extract the S3 specific logic out from JournalWriter

### PR DESCRIPTION
**Pull request by xgong**
https://alluxio.atlassian.net/browse/ALLUXIO-1509

### CODE REVIEW

**Comment by xgong** [comment]
Jenkins, test this please.

**Comment by uronce-cc** [comment]
LGTM

**Comment by apc999** [comment]
@calvinjia this PR introduces UFS API change, could you take a look?

**Comment by calvinjia** [comment]
alluxio-bot, check this please.

File: core/common/src/main/java/alluxio/underfs/UnderFileSystem.java [hunk 1]
```diff
@@ -706,4 +706,13 @@ public void setProperties(Map<String, String> properties) {
    * @throws IOException if a non-Alluxio error occurs
    */
   public abstract short getMode(String path) throws IOException;
+
+  /**
+   * Whether this type of UFS needs the synchronous journaling.
+   *
+   * @return true if this type of UFS needs the synchronous journaling, false otherwise
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/underfs/UnderFileSystem.java]: Maybe a better term is `supportsFlush`?

**Comment by calvinjia** [comment]
@xgong Thanks for the improvement, could you also apply the `true` flag to S3A and GCS under storages?

**Comment by gpang** [comment]
@xgong @calvinjia I think `supportFlush` might be misleading. The reason object stores need to rotate the log is because the earlier call to `flush` call doesn't actually flush to the object store. The log rotation will close the file and create a new one. I think the method should be renamed, or the returned values should be flipped.

What do you think?

**Comment by calvinjia** [comment]
@xgong Could you switch the true/false returns as @gpang mentioned and rename the method to `supportsFlush` for consistency with our other method names? Otherwise this change looks good to me.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**